### ISSUE TITLE
Resume Previous Training

### ISSUE THREAD

**Issue by JiaLim98** [zylo117/Yet-Another-EfficientDet-Pytorch#390]
Dear zylo117,

I have managed to train for one cycle (from the beginning until the end of train.py). However, I encountered a problem about continuing previous training session. I stopped my training at 15epoch full (meaning I did not stop the training midway), but was unable to resume after that.

Normal training console:
![image](https://user-images.githubusercontent.com/66768413/84597666-1f343a80-ae98-11ea-9232-887edc499803.png)

I tried using "-w last" but it did not work. I moved the computed weights at epoch 15 to the weights folder and direct the weights to that file give me the same outcome. It just stops at "[Info] freezed backbone" and the execution of train.py is completed. Can you let me know how to solve it?

One more question: If the visualization code in the tutorial did not plot anything, it means that my trained model is detecting nothing on it. Is that caused by insufficient number of epochs in training and the anchors set is not suitable?

Cheers,
JiaLim98 (newbie in deep learning)

**Comment by zylo117** [comment]
Is it the latest code?
Try running the tutorial on colab, it should work.
If the tutorial can't work as expected, you should check your environment.

**Comment by JiaLim98** [comment]
Yes, it is the latest code and I am running on Colab.
If you think there shouldn't be a problem and since no one has the same problem. Then I think I might have changed other something accidentally.
Thank you for the fast reponse!

**Comment by zylo117** [comment]
Maybe clean up everything and re-run the tutorial would help.

**Comment by JiaLim98** [comment]
I will now get the latest code again and start again.

This is what I see when I try to continue previous training session on Colab.
![image](https://user-images.githubusercontent.com/66768413/84629767-6dd7e800-af1d-11ea-898e-ad227bbb01ee.png)
Then it just ends there. Am I missing anything?

**Comment by zylo117** [comment]
I see. This is absolutely ok!
You just need to specify a greater num_epochs...
Current epoch is already 99, num_epoch 1 means stop at epoch 1...

**Comment by JiaLim98** [comment]
I see. This is awkward... I thought I just have to specify how many more epochs I need. So sorry for inconvenience caused.

I will now try solve the visualization code so that it can plot my bboxes on my own data. From the previous threads, I see some with boxes that are wrong but mine was just zero output like nothing detected. I looked into the val_bbox_results.json file, there are many bbox recorded in one image. It just won't show in the visualization code. Nevertheless, I am trying the tutorial again see what did I missed.

**Comment by JiaLim98** [comment]
Hi Zylo117,

I realised my accuracy has not converge for now. The visualization code in tutorial is still not able to plot any bbox on the image I specified. I guess it is due to the low accuracy of mine. I have used 7 anchors computed from Cli98. Here is my result at 200epochs for D0 (obviously I need more epochs).
![image](https://user-images.githubusercontent.com/66768413/84910109-0d95a180-b0e9-11ea-99a5-02fc3d216649.png)
I have some questions to ask you:
1. You mentioned in #397, to replace the coco_eval.py with the original cocoeval.py. I understand that the negative one is because the model trained fails to detect an object with size larger than 96x96. Is it necessary to use cocoeval.py? Because I find using coco_eval.py is more convenient.

2. I am doing my training using Google Colab which utilizes Tesla K80 as the GPU. Did you use Colab to train before? If yes, what is the value you used for batch size and num workers? I have a dataset of 10,668 (size 600x600) images.

3. If I increase the val_interval and save_interval, will it decrease the training duration by alot? Or even better, can the save_interval be set to zero? Because I find the weights file (.pth) is sufficient for me to continue previous training.

4. When reaching final stages, you said to decrease the learning rate and change the optimizer to 'sgd'. What do you mean by 'final stages'? Is it after the the accuracy converges?

Regards,
JiaLim98

**Comment by zylo117** [comment]
What do you by tutorial? The script or the tutorial itself?
What exactly is the problem? You can't train well on your custom dataset or you can't train well on shape dataset?

**Comment by JiaLim98** [comment]
I can train on my dataset, no problem. Just that when I use the visualize code from the tutorial to plot my results, there's nothing shown. 
The tutorial totally has no problem.
Do you mind sharing the answers to the four questions? It would be great! :)

**Comment by zylo117** [comment]
try this repo to generate anchors, it helped me. https://github.com/mnslarcher/kmeans-anchors-ratios

In this case, maybe you should set suitable anchors.
And it'd be better not to modify the numbers of anchors. Did you mean 7 anchors ratios? It is just too much, right? Maybe just make it 3, like the default ones.

**Comment by JiaLim98** [comment]
I used https://github.com/mnslarcher/kmeans-anchors-ratios and realised the scale he used are [32 64 128 256 512]. That is huge compared to the default one which are
![image](https://user-images.githubusercontent.com/66768413/85001983-ebeaf780-b186-11ea-8a95-c9f505f5a745.png)
This is what I get after using that repo:
![image](https://user-images.githubusercontent.com/66768413/85002036-01602180-b187-11ea-8362-827383fb5e94.png)
Am I putting it in correctly? The loss calculated when training was in hundred thousands. The losses calculated when using the default one was below one hundred.

**Comment by zylo117** [comment]
those scales are not the same, don't change them
